rust/lancedb/src/query.rs

```diff
@@ -294,6 +324,51 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_select_with_transform() {
+        let batches = make_non_empty_batches();
+        let ds = Dataset::write(batches, "memory://foo", None).await.unwrap();
+
+        let ds = DatasetConsistencyWrapper::new_latest(ds, None);
+
+        let query = Query::new(ds.clone()).nearest_to(&[0.1; 4]);
+        let result = query.limit(10).filter("id % 2 == 0").execute_stream().await;
+        let mut stream = result.expect("should have result");
+        // should only have one batch
+        while let Some(batch) = stream.next().await {
+            // post filter should have removed some rows
+            assert!(batch.expect("should be Ok").num_rows() < 10);
+        }
+
+        let query = Query::new(ds).nearest_to(&[0.1; 4]);
```

Comment:
Any particular reason to do `nearest_to`/`limit` instead of a plain query?  This is just a nitpick to keep the tests as simple as possible.